client/src/main/java/sprite/Sprite.java

```diff
@@ -0,0 +1,29 @@
+package sprite;
+
+import engine.geometry.Vector;
+import engine.object.GameObject;
+
+import java.awt.*;
+
+public class Sprite {
+
+    private final Vector position;
+    private final Image image;
+
+    Sprite(Image image, GameObject gameObject) {
```

Comment:
You are only required gameObject.position. No need to pass whole object right now

Sprite(Vector position, Image image)